application/src/main/java/org/thingsboard/server/actors/app/AppActor.java

```diff
@@ -123,7 +124,11 @@ private void initTenantActors() {
         try {
             if (systemContext.isTenantComponentsInitEnabled()) {
                 PageDataIterable<Tenant> tenantIterator = new PageDataIterable<>(tenantService::findTenants, ENTITY_PACK_LIMIT);
+                Set<UUID> assignedProfiles = systemContext.getServiceInfoProvider().getAssignedTenantProfiles();
                 for (Tenant tenant : tenantIterator) {
+                    if (!assignedProfiles.isEmpty() && !assignedProfiles.contains(tenant.getTenantProfileId().getId())) {
```

Comment:
Please take a look at ```onToDeviceActorMsg```. We don't need to dispatch those messages for tenants processed on different servers. Otherwise, the TenantActor will be created. See all usages of  getOrCreateTenantActor

Reply:
Much easier to read is 

```java
if (assignedProfiles.isEmpty() || assignedProfiles.contains(tenant.getTenantProfileId().getId())) {
    log.debug("[{}] Creating tenant actor", tenant.getId());
    getOrCreateTenantActor(tenant.getId());
    log.debug("[{}] Tenant actor created.", tenant.getId());
}

```

Reply:
Agree. The conditions with NOT is hard to follow